File: CS101_Eclipse/apt_friendscore/src/FriendScore.py
'''
Created on Mar 27, 2013

@author: Jimmy
'''
def highestScore(friends):
    twofriends = 0
    for i in range(0,len(friends)):
        dummy = []
        for j in range(0,len(friends)):
            if friends[i][j] == 'Y':
                dummy.append(j)    
                for k in range(0,len(friends)):
                    if friends[j][k] == 'Y':
                        dummy.append(k);
        twofriends = max(len(set(dummy)) - 1, twofriends)
    return twofriends;

File: CS101_Eclipse/APT 3/src/FriendScore.java
import java.util.*;

public class FriendScore {

	public int highestScore(String[] friends) {
		int twofriends = 0;

		for (int i = 0; i < friends.length; i++){

			Set<Integer> set = new HashSet<Integer>();

			for (int j = 0; j < friends.length; j++){
				if (friends[i].charAt(j) == 'Y'){
					set.add(j);	

					for (int k = 0; k < friends.length; k++){
						if (friends[j].charAt(k) == 'Y'){
							set.add(k);
						}
					}
				}
			}
			twofriends = Math.max(set.size() - 1, twofriends);
		}
		return twofriends;
	}
}
